build.gradle

```diff
@@ -82,17 +82,13 @@ subprojects {
     }
 
     dependencies {
-        // This adds constraints (which can be overriden below) for all dependencies egeria uses
+        // This adds constraints (which can be overridden below) for all dependencies egeria uses
         implementation platform("org.odpi.egeria:egeria:${egeriaVersion}")
 
         // specifies versions if dependencies used in projects
         constraints
                 {
                     implementation("ch.qos.logback:logback-classic:${logbackVersion}")
-                    implementation("org.odpi.egeria:audit-log-framework:${egeriaVersion}")
```

Comment:
These are just removing 'constraints' not dependencies. 
Constraints mean that IF a dependency is used, then this is the version we try to resolve too.

In the xtdb connector, we also have a multi-module project, and further constraints effectively apply from the 'allProjects' and 'subProjects' sections of the top level build.gradle.

In there we have
```
         implementation platform("org.odpi.egeria:egeria:${egeriaVersion}")
```

which uses the new gradle 'BOM' to align dependencies with those of core egeria.

It's therefore appropriate to try not to have additional constraints where we can reasonably expect egeria to pin them. This probably includes jackson for example.

Where a constraint exists and the dependency is needed, the version can be left unspecified. In fact if no constraint is then found, gradle will complain.

Reply:
We have lucene included. it's possible this could cause a clash if luceneVersion doesn't match the base.

I suspect the best option here would be to make use of the BOM definitions - both lucene-core and lucene-queryparser are already in that list, so remove them here (as with jackson).

So leave as implementation (as we want to compile & run) but remove the version component

Same applies to lucene-analyzers common

Consider what to do with kafka - at least it's compile only, so effect is limited

I think this discussion again re-enforces a need to use a dedicated classpath for loading these connector libraries, which probably would avoid these issues - both between tech but also version changes which make the current mechanism very fragile

Reply:
Cleaned up constraints